mesonbuild/compilers/detect.py

```diff
@@ -171,7 +171,14 @@ def detect_static_linker(env: 'Environment', compiler: Compiler) -> StaticLinker
             trials = [defaults['gcc_static_linker']] + default_linkers
         elif compiler.id == 'clang':
             # Use llvm-ar if available; needed for LTO
-            trials = [defaults['clang_static_linker']] + default_linkers
+            llvm_ar = defaults['clang_static_linker']
+            extra_linkers = []
+            exe = compiler.name_string()
+            suffix = exe.rsplit('-')[-1]
```

Comment:
I think this will probably incorrectly handle things such as `CC=/home/eschwartz/bin/clang-wrapper`.

Another issue might be `CC="clang -target=i686-pc-linux-gnu"` as I believe this is part of the name_string too.

Maybe a better option here is `compiler.version` trimmed to the first component.

Reply:
That's going to be awkward to handle then as the goal here is to get the compiler binary name, check if it's suffixed, and only suffix `llvm-ar` if it is - so if the compiler invocation is just `clang`, we should use `llvm-ar`, if it's `clang-18`, then `llvm-ar-18`. We'll have a good long think on how best to handle that with this extra info.

Reply:
Another option would be to check if `any(s.endswith(f'-{major}') for s in compiler
exelist)`

Reply:
We've built a provisional patch for this, incorporating the feedback from the next review item too. Please give it a look and let us know if it looks good to you. If it does, we'll rebase it in to the patchset properly and also fix the same problem in environment.py - compute_llvm_suffix() - which is afflicted the same way.

Reply:
Just checking in on this as we'd appreciate a review of the fixup we've done to verify it's suitable and the right direction before we go and put the work in to do the same thing in compute_llvm_suffix() for environment.py

Reply:
Your fixup commit looks fine, so yes -- feel free to use that version in both places.